### PR TITLE
provider/azure: Added affinity group resource.

### PR DESCRIPTION
**Pull request by aznashwan**
Added support, tests and docs for Azure affinity groups.

### CODE REVIEW

File: builtin/providers/azure/resource_azure_affinity_group.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+// resourceAzureAffinityGroupExists does all the necessary API calls to
+// check for the existence of the affinity group on Azure.
+func resourceAzureAffinityGroupExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	affinityGroupClient := meta.(*Client).affinityGroupClient
+
+	log.Println("[INFO] Issuing Azure Affinity Group get request.")
+	name := d.Get("name").(string)
+	_, err := affinityGroupClient.GetAffinityGroup(name)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
```
Comment (phinze) [inline builtin/providers/azure/resource_azure_affinity_group.go]: How nice of the client to provide this function to us! :grinning:

File: website/source/docs/providers/azure/r/affinity_group.html.markdown [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Creates a new affinity group on Azure with its own .cloudapp.net domain.
+
+## Example Usage
+
+```
+resource "azure_affinity_group" "terraform-service" {
+	name = "terraform-group"
+	location = "North Europe"
+    label = "tf-group-01"
```
Comment (phinze) [inline website/source/docs/providers/azure/r/affinity_group.html.markdown]: Indentation is off here

File: website/source/docs/providers/azure/r/affinity_group.html.markdown [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+The following arguments are supported:
+
+* `name` - (Required) The name of the affinity group. Must be unique on your
+    Azure subscription.
+
+* `location` - (Required) The location where the affinity group should be created.
+    For a list of all Azure locations, please consult [this link](http://azure.microsoft.com/en-us/regions/).
+
+* `label` - (Label) A label to be used for tracking purposes. Must be
+    non-void.
```
Comment (phinze) [inline website/source/docs/providers/azure/r/affinity_group.html.markdown]: What does non-void mean in this context. Looks like it just means "Required" in which case we can just drop the sentence and change the parens from `(Label)` to `(Required)`

**Comment by phinze** [comment]
A few tiny tiny nits, but this looks great, @aznashwan!

File: website/source/docs/providers/azure/r/affinity_group.html.markdown (hunk 3 above)
Comment (aznashwan) [inline website/source/docs/providers/azure/r/affinity_group.html.markdown]: Brain fart of the century right there :disappointed:

**Comment by phinze** [comment]
Beautiful - great work once again!

**Pull request merged**